### PR TITLE
[Mobile] - Phase Result Clean Up

### PR DESCRIPTION
**Pull request by drewjhart**
Phase Results has been rewritten to remove redundant variables and to ensure that the calculation of scores is wrapped in `useFocusEffect` and `useCallback` so they are only run when the screen is shown.

### CODE REVIEW

File: mobile/src/Student/screens/PhaseResult/index.js [hunk 1]
```diff
@@ -7,167 +7,104 @@ import TeamFooter from '../../../components/TeamFooter'
...
-
-        if (selectedTrickAnswer?.text === answer.text) {
-            setPhase2Score(percentage)
-        } else {
-            setPhase2Score(0)
+    
+        const getAnswer = ({item, index}) => {
+          return <Answer
+            icon={teamAvatar.smallSrc}
+            text={`${alphabets[index]}. ${item.text}`}
```
Comment (maniramezan) [inline mobile/src/Student/screens/PhaseResult/index.js:46]: Wondering if we should add a `title` or `orderedText` to question answers and do this logic there. Definitely a work for a follow up PR, but that helps with not duplicating this logic on Host and mobile and make sure both show the same order and alphabet letter

File: mobile/src/Student/containers/GameSessionContainer.js [hunk 2]
```diff
@@ -161,6 +158,8 @@ const GameSessionContainer = ({ children }) => {
                           )
                           return    
                       }
+                      const newAnswers = team.teamMembers[0].answers.concat(teamAnswer)
```
[the remark this answers is not in the document]
Comment (drewjhart) [reply mobile/src/Student/containers/GameSessionContainer.js:161]: Thanks for the review. When initially created, answers is an empty array but I've rewritten this to check the array for empty/null and then respond accordingly.

File: mobile/src/Student/screens/PhaseResult/index.js [hunk 3]
```diff
@@ -7,167 +7,104 @@ import TeamFooter from '../../../components/TeamFooter'
...
+        const getP1AnswerMode = (choiceText) =>{
+          if (choiceText === correctAnswer.text) 
+            return AnswerMode.RightAnswer
+          else if (choiceText === selectedAnswer.text) 
+            return AnswerMode.UserAnswer
+          else
+            return AnswerMode.ShowEmptyRightIcon
+        } 
+    
+        const getP2AnswerMode = (choiceText) => {
```
[the remark this answers is not in the document]
Comment (drewjhart) [reply mobile/src/Student/screens/PhaseResult/index.js]: Good call, fixed

File: mobile/src/Student/screens/PhaseResult/index.js (hunk 1 above)
Comment (drewjhart) [reply mobile/src/Student/screens/PhaseResult/index.js:47]: Yeah, that makes a lot of sense. I've created https://github.com/rightoneducation/righton-app/issues/567 to track this issue.

**Pull request merged**